rita_common/src/network_monitor/mod.rs

```diff
@@ -102,84 +83,62 @@ pub struct IfaceStats {
     packet_loss: PacketLossStats,
 }
 
-impl Message for GetStats {
-    type Result = Result<Stats, RitaCommonError>;
-}
-
 pub type Stats = HashMap<String, IfaceStats>;
 
-impl Handler<GetStats> for NetworkMonitor {
-    type Result = Result<Stats, RitaCommonError>;
-
-    fn handle(&mut self, _msg: GetStats, _ctx: &mut Context<Self>) -> Self::Result {
-        let mut stats = Stats::new();
-
-        for (iface, latency_stats) in self.latency_history.iter() {
-            if let Some(packet_loss_stats) = self.packet_loss_history.get(iface) {
-                stats.insert(
-                    iface.clone(),
-                    IfaceStats {
-                        latency: LatencyStats {
-                            avg: latency_stats.get_avg(),
-                            std_dev: latency_stats.get_std_dev(),
-                        },
-                        packet_loss: PacketLossStats {
-                            avg: packet_loss_stats.get_avg(),
-                            five_min_avg: packet_loss_stats.get_five_min_average(),
-                        },
+pub fn get_stats(_msg: GetStats) -> Stats {
+    let mut stats = Stats::new();
+
+    for (iface, latency_stats) in (*(NETWORK_MONITOR.write().unwrap())).latency_history.iter() {
+        if let Some(packet_loss_stats) = (*(NETWORK_MONITOR.write().unwrap())).packet_loss_history.get(iface) {
+            stats.insert(
+                iface.clone(),
+                IfaceStats {
+                    latency: LatencyStats {
+                        avg: latency_stats.get_avg(),
+                        std_dev: latency_stats.get_std_dev(),
                     },
-                );
-            } else {
-                error!("Found entry in one that's not in the other ")
-            }
+                    packet_loss: PacketLossStats {
+                        avg: packet_loss_stats.get_avg(),
+                        five_min_avg: packet_loss_stats.get_five_min_average(),
+                    },
+                },
+            );
+        } else {
+            error!("Found entry in one that's not in the other ")
         }
-
-        Ok(stats)
     }
+    stats
 }
 
 pub struct GetNetworkInfo;
 
-impl Message for GetNetworkInfo {
-    type Result = Result<NetworkInfo, RitaCommonError>;
-}
-
-impl Handler<GetNetworkInfo> for NetworkMonitor {
-    type Result = Result<NetworkInfo, RitaCommonError>;
-
-    fn handle(&mut self, _msg: GetNetworkInfo, _ctx: &mut Context<Self>) -> Self::Result {
-        match self.last_babel_dump.clone() {
-            Some(dump) => Ok(dump),
-            None => Err(RitaCommonError::MiscStringError(
-                "No babel info ready!".to_string(),
-            )),
-        }
+pub fn get_network_info(_msg: GetNetworkInfo) -> Result<NetworkInfo, RitaCommonError> {
+    match (*(NETWORK_MONITOR.write().unwrap())).last_babel_dump.clone() {
```

Comment:
this is a read only operation, it does not need a write lock, only a read lock. It's much better to use a read lock than a write lock when possible becuase there can be an unlimited number of parallel readers but only one writer.